stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -856,3 +856,66 @@ for func in (:log, :sqrt)
         end
     end
 end
+
+"""
+    symmetricpart(A, uplo=:U)
+
+Return the symmetric part of the square matrix `A`, defined as `(A + transpose(A)) / 2`,
+as a [`Symmetric`](@ref) matrix.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+symmetricpart(A::AbstractMatrix{T}, uplo=:U) where {T} =
+    symmetricpart!(copyto!(similar(A, typeof(one(T) / 2)), A), uplo)
+symmetricpart(x::Number) = x
+
+"""
+    symmetricpart!(A, uplo=:U)
+
+Overwrite the square matrix `A` with its symmetric part, `(A + transpose(A)) / 2`,
+and return `Symmetric(A, uplo)`.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+symmetricpart!(A::AbstractMatrix, uplo::Symbol=:U) =
+    Symmetric(_hermorsympart!(symmetricpart, transpose, A, char_uplo(uplo)), uplo)
+
+"""
+    hermitianpart(A, uplo=:U)
+
+Return the Hermitian part of the square matrix `A`, defined as `(A + A') / 2`,
+as a [`Hermitian`](@ref) matrix.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+hermitianpart(A::AbstractMatrix{T}, uplo=:U) where {T} =
+    hermitianpart!(copyto!(similar(A, typeof(one(T) / 2)), A), uplo)
+hermitianpart(x::Number) = real(x)
+
+"""
+    hermitianpart!(A, uplo=:U)
+
+Overwrite the square matrix `A` with its Hermitian part, `(A + A') / 2`,
+and return `Hermitian(A, uplo)`.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+hermitianpart!(A::AbstractMatrix, uplo::Symbol=:U) =
+    Hermitian(_hermorsympart!(hermitianpart, adjoint, A, char_uplo(uplo)), uplo)
+
+function _hermorsympart!(real::Function, conj::Function, A::AbstractMatrix, uplo::Char)
+    require_one_based_indexing(A)
+    n = checksquare(A)
+    triangle = uplo === 'U' ? (j -> 1:(j - 1)) : (j -> (j + 1):n)
+    @inbounds for j in 1:n
+        A[j, j] = real(A[j, j])
+        for i in triangle(j)
```

Comment:
Maybe just
```
for i in (uplo === 'U' ? (1:(j - 1)) : ((j + 1):n))
```
?  I'm skeptical that a function call is faster than a conditional, here, since the compiler presumably won't be able to inline `triangle(j)`.